src/v0/destinations/mp/util.js

```diff
@@ -301,6 +302,46 @@ function trimTraits(traits, contextTraits, setOnceProperties) {
   };
 }
 
+/**
+ * Generates a custom event name for a page or screen.
+ *
+ * @param {Object} message - The message object
+ * @param {string} userDefinedEventTemplate - The user-defined event template to be used for generating the event name.
+ * @throws {ConfigurationError} If the event template is missing.
+ * @returns {string} The generated custom event name.
+ * @example
+ * const userDefinedEventTemplate = "Viewed {{ category }} {{ name }} Page";
+ * const message = {name: 'Home', properties: {category: 'Index'}};
+ * output: "Viewed Index Home Page"
+ */
+const generatePageOrScreenCustomEventName = (message, userDefinedEventTemplate) => {
+  if (!userDefinedEventTemplate) {
+    throw new ConfigurationError(
+      'Event name template is not configured. Please provide a valid value for the `Page/Screen Event Name Template` in the destination dashboard.',
+    );
+  }
+
+  let eventName = userDefinedEventTemplate;
+
+  if (isDefinedAndNotNull(message.properties?.category)) {
+    // Replace {{ category }} with actual values
+    eventName = eventName.replace(/{{\s*category\s*}}/g, message.properties.category);
+  } else {
+    // find {{ category }} surrounded by whitespace characters and replace it with a single whitespace character
+    eventName = eventName.replace(/\s{{\s*category\s*}}\s/g, ' ');
+  }
+
+  if (isDefinedAndNotNull(message.name)) {
+    // Replace {{ name }} with actual values
+    eventName = eventName.replace(/{{\s*name\s*}}/g, message.name);
+  } else {
+    // find {{ name }} surrounded by whitespace characters and replace it with a single whitespace character
+    eventName = eventName.replace(/\s{{\s*name\s*}}\s/g, ' ');
+  }
+
+  return eventName.trim();
```

Comment:
We should not trim any leading or trailing whitespaces.

Reply:
This was not the case in device mode, Fixed it cloud mode.